Source/RxCocoa/UIViewPropertyAnimator+Rx.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  UIViewPropertyAnimator+Rx.swift
+//  RxSwiftExt
+//
+//  Created by Wittemberg, Thibault on 29/03/18.
+//  Copyright © 2017 RxSwift Community. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import RxSwift
+import RxCocoa
+
+@available(iOSApplicationExtension 10.0, *)
+extension Reactive where Base == UIViewPropertyAnimator {
+
+    /// Completable that when subscribed to, starts the animation
+    /// and completes once the animation is ended
+    public var animate: Completable {
+        return Completable.create(subscribe: { (completable) -> Disposable in
+
+            self.base.addCompletion({ (position) in
+                if position == .end {
+                    completable(.completed)
+                }
+            })
+
+            self.base.startAnimation()
+
+            return Disposables.create {
+                self.base.stopAnimation(true)
+            }
+        })
+    }
+
+    /// Completable that when subscribed to, starts the animation after a delay
+    /// and completes once the animation is ended
+    ///
+    /// - Parameter delay: the delay to apply to the animation start
+    /// - Returns: the Completable that will send .completed once the animation is ended
+    public func animate(afterDelay delay: Double) -> Completable {
+        return Completable.create(subscribe: { (completable) -> Disposable in
+
+            self.base.addCompletion({ (position) in
+                if position == .end {
+                    completable(.completed)
+                }
+            })
+
+            self.base.startAnimation(afterDelay: delay)
+
+            return Disposables.create {
+                self.base.stopAnimation(true)
+            }
+        })
+    }
+}
+
+@available(iOSApplicationExtension 10.0, *)
+extension UIViewPropertyAnimator {
```

Comment:
Why is this needed? I think since UIViewPropertyAnimator is NSObject, you get this out-of-the-box.

Reply:
done.